tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,496 @@
+from liota.lib.utilities.utility import ordered_list
+
+# Create a pint unit registry
+ureg = pint.UnitRegistry()
+
+
+# Monkey patched init method of
+def mocked_init_mqtt_dcc_comms(self, *args, **kwargs):
+    pass
+
+
+# Sampling function
+def sampling_function():
+    pass
+
+
+class MQTTTest(unittest.TestCase):
+    """
+    AWSIoT unit test cases
+    """
+
+    def setUp(self):
+        """
+        Method to initialise the AWSIoT parameters.
+        :return: None
+        """
+
+        # EdgeSystem name
+        self.edge_system = Dell5KEdgeSystem("TestEdgeSystem")
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        self.mocked_mqtt_dcc_comms = MqttDccComms()
+
+        self.aws = AWSIoT(self.mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+    def tearDown(self):
+        """
+        Method to cleanup the resource created during the execution of test case.
+        :return: None
+        """
+        self.edge_system = None
+        self.mocked_mqtt_dcc_comms = None
+        self.aws = None
+
+    def test_aws_class_implementation(self):
+        """
+        Method to test the implementation of AWSIoT class.
+        :return: None
+        """
+
+        self.assertIsInstance(self.aws, AWSIoT, "Check the implementation of AWSIoT class")
+
+    def test_validation_of_comms_parameter(self):
+        """
+        Method to test the validation of AWSIoT class for invalid connections object.
+        :return: None
+        """
+        # Checking whether implementation raising the TypeError Exception for invalid comms object
+        with self.assertRaises(TypeError):
+            AWSIoT("Invalid object", enclose_metadata=True)
+
+    def test_implementation_register_entity(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Register the edge
+        registered_entity = self.aws.register(self.edge_system)
+
+        # Check the returned object is of the class RegisteredEntity
+        self.assertIsInstance(registered_entity, RegisteredEntity)
+
+    def test_implementation_register_metric(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = self.aws.register(test_metric)
+
+        # Check the returned object is of the class RegisteredMetric
+        self.assertIsInstance(registered_metric, RegisteredMetric)
+
+    def test_implementation_create_relationship(self):
+        """
+        Method to test RegisteredEntity and Parent and RegisteredMetric as child.
```

Comment:
Test case to check RegisteredEntity as Parent and RegisteredMetric as Child.  Sorry it was a typo when I gave review.

This is a `test case`, that is why I suggested use test case instead of method.  Please change this in all occurrences.